runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -0,0 +1,165 @@
+package org.corfudb.runtime;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.io.FileUtils;
+import org.corfudb.protocols.logprotocol.OpaqueEntry;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.util.serializer.Serializers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+@Slf4j
+public class Restore {
+
+    // The path of backup tar file
+    private final String filePath;
+
+    // The path of a temporary directory under which the unpacked table's backup files are stored
+    private final String restoreTempDirPath;
+
+    // The stream IDs of tables which are restored
+    private List<UUID> streamIDs;
+
+    // The Corfu Store associated with the runtime
+    private CorfuStore corfuStore;
+
+    /**
+     * Unpacked files from backup tar file are stored under RESTORE_TEMP_DIR. They are deleted after restore finishes.
+     */
+    private static final String RESTORE_TEMP_DIR_PREFIX = "corfu_restore_";
+
+    /***
+     * @param filePath      - the path of backup tar file
+     * @param runtime       - the runtime which is performing the restore
+     * @throws IOException when failed to create the temp directory
+     */
+    public Restore(String filePath, CorfuRuntime runtime) throws IOException {
+        this.filePath = filePath;
```

Comment:
we're restoring whatever is available in that TAR... imagine someone manipulates that TAR and removes some files... should we have a validation similar to the case of Backup, where if a list of streams to restore is given, or tagged streams is specified as preferred option, at the end we consider successful, if and only if we have restored everything requested in the API.

Reply:
The requirement for restore is to restore whatever inside the given TAR file. We used to let users input a list of tables they want to restore in the restore API but Graham's team said they don't really need that feature. That's why we remove that parameter and just restore whatever they give us.